### ISSUE TITLE
expired transaction not in correct V4 form makes zcashd repeatedly rescan

### ISSUE THREAD

**Issue by arielgabizon** [zcash/zcash#3584]
Say you got a transaction to you 2.0.0 wallet that turned out to have bad expiry
(in my case cause I got it from the 2.0.0 faucet that was not caught up with the blockchain)

When you run 2.0.1 because of the V4 format change you'll get
"Warning: Warning: error reading wallet.dat! All keys read correctly, but transaction data or address book entries might be missing or incorrect."
and the wallet will rescan the blockchain.

The problem is that when this transaction is expired, this transaction will neither be deleted nor fixed to the 2.0.1 V4 disk format.
So you'll keep getting a warning and rescan each time you restart 2.0.1 zcashd.

The way to resolve this as @bitcartel showed me is to run
`zcashd -zapwallettxes`

that does delete the expired tx.

Related to https://github.com/zcash/zcash/issues/3393

**Comment by str4d** [comment]
Yep, this is another kind of transaction that gets tripped up by the format change. The other that we knew about being transactions created with a Sapling key that is then removed from the wallet. We never remove keys from the wallet in normal operation, so in practice no users will run into that one. Expired transactions that are in the wallet but not in the chain, OTOH, are possible for users to run into, because #3393 has not been fixed.

I still think this is something we can just tell users to use `zcashd -zapwallettxes` to fix, however.

**Comment by arielgabizon** [comment]
Wondering if we should make the default tx expiry infinity

**Comment by daira** [comment]
Well, we might be able to fix this before the release. The point of the RC is to find things like this.
